scripts/gas-refund-program/token-pricing/psp-chaincurrency-pricing.ts

```diff
@@ -40,32 +49,37 @@ export async function fetchDailyPSPChainCurrencyRate({
     }),
   ]);
 
-  assert(dailyAvgChainCurPrice.length > 0, 'could not find any rate');
+  const kdailyAvgChainCurPrice = Object.keys(dailyAvgChainCurPrice).length;
+  const kdailyAvgPspPrice = Object.keys(dailyAvgPspPrice).length;
+  assert(kdailyAvgChainCurPrice > 0, 'could not find any rate');
   assert(
-    dailyAvgChainCurPrice.length === dailyAvgPspPrice.length,
-    `Invalid price length got: ${dailyAvgChainCurPrice.length} and ${dailyAvgPspPrice.length}`,
+    kdailyAvgChainCurPrice === kdailyAvgPspPrice,
+    `Invalid price length got: ${kdailyAvgChainCurPrice} and ${kdailyAvgPspPrice}`,
   );
+  logger.info(`Successfully retrieved ${kdailyAvgChainCurPrice} prices`);
 
-  logger.info(`Successfully retrieved ${dailyAvgChainCurPrice.length} prices`);
-
-  const dailyAvgPSPToChainCurPrice = dailyAvgChainCurPrice.map(
-    (chainCurPrice, i) => ({
-      timestamp: chainCurPrice.timestamp,
-      rate: dailyAvgPspPrice[i].rate / chainCurPrice.rate,
-    }),
-  );
+  const combinedPrices = Object.keys(
+    dailyAvgChainCurPrice,
+  ).reduce<PricesByTimestamp>((acc, timestamp) => {
+    const pspPrice = dailyAvgPspPrice[timestamp];
+    const chainPrice = dailyAvgChainCurPrice[timestamp];
```

Comment:
Do timestamps always match for different coins? They probably do

Reply:
They do here because timestamp have been projected to "start of the day" here on averaging. But the right way to do it is [there](https://github.com/paraswap/paraswap-volume-tracker/pull/29)